util/src/main/java/io/kubernetes/client/util/KubeConfig.java

```diff
@@ -214,6 +222,60 @@ public String getAccessToken() {
         }
       }
     }
+    Object exec = currentUser.get("exec");
```

Comment:
Please break this out into a separate method.

Actually, I think it would be good like this:

`getExecAuthProviderToken()`

```java
public String getExecAuthProviderToken(...) {
   String command = getAuthCommand(...);
   Object response = runCommand(command);
   return extractToken(response);
}
```

Or something like that.

Reply:
> Please break this out into a separate method.

Yes, already noted as per comment two lines down.